packages/rocketchat-ldap/server/sync.js

```diff
@@ -84,14 +85,19 @@ getDataToSyncUserData = function getDataToSyncUserData(ldapUser, user) {
 					}
 					break;
 
-				case 'name':
+				default:
+					if (!_.find(whitelistedUserFields, (el) => el === userField.split('.')[0])) {
+						logger.debug(`user attribute not whitelisted: ${ userField }`);
+						return;
+					}
+
 					const tmpLdapField = RocketChat.templateVarHandler(ldapField, ldapUser.object);
+					const userFieldValue = _.reduce(userField.split('.'), (acc, el) => acc[el], user);
```

Comment:
What happens when a subfield is undefined? Like `customFields.name.first` and `name` is undefined?

Reply:
@rodrigok I see you've already merged this code into develop branch, but I think you were right regarding this part of the code. An exception would be thrown and it would not be properly handled. You were faster than me, I didn't have enough time to push the changes before your merge. I'd recommend replacing this:

[`const userFieldValue = _.reduce(userField.split('.'), (acc, el) => acc[el], user);`](https://github.com/RocketChat/Rocket.Chat/blob/8e998c00d11a04564145346728e612e119b09824/packages/rocketchat-ldap/server/sync.js#L95)

by this (or something more elegant...):

```javascript
let userFieldValue;

try {
	userFieldValue = _.reduce(userField.split('.'), (acc, el) => acc[el], user);
	if (!userFieldValue) {
		throw new Error();
	}
} catch (err) {
	logger.debug(`user attribute does not exist: ${ userField }`);
	return;
}
```

What do you think @rodrigok?